lib/internal/blob.js

```diff
@@ -188,49 +228,143 @@ class Blob extends JSTransferable {
 
     const span = MathMax(end - start, 0);
 
-    return new InternalBlob(
-      this[kHandle].slice(start, start + span), span, contentType);
+    return createBlob(
+      this[kHandle].slice(start, start + span),
+      span,
+      contentType);
   }
 
-  async arrayBuffer() {
+  /**
+   * @returns {Promise<ArrayBuffer>}
+   */
+  arrayBuffer() {
+    if (!isBlob(this))
+      return PromiseReject(new ERR_INVALID_THIS('Blob'));
+
+    // If there's already a promise in flight for the content,
+    // reuse it, but only once. After the cached promise resolves
```

Comment:
Not Maybe I'm misunderstanding what you meant, but it seems the promise can be re-used infinitely many times until it has fulfilled.
```suggestion
    // If there's already a promise in flight for the content,
    // reuse it. After the cached promise resolves
```